flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -528,6 +538,71 @@ public void triggerSavepoint(
         }
     }
 
+    public Optional<Savepoint> getLastCheckpoint(JobID jobId, Configuration conf) throws Exception {
+        try (RestClusterClient<String> clusterClient =
+                (RestClusterClient<String>) getClusterClient(conf)) {
+
+            var headers = CustomCheckpointingStatisticsHeaders.getInstance();
+            var params = headers.getUnresolvedMessageParameters();
+            params.jobPathParameter.resolve(jobId);
+
+            CompletableFuture<CheckpointHistoryWrapper> response =
+                    clusterClient.sendRequest(headers, params, EmptyRequestBody.getInstance());
+
+            var checkpoints =
+                    response.get(
+                            configManager
+                                    .getOperatorConfiguration()
+                                    .getFlinkClientTimeout()
+                                    .getSeconds(),
+                            TimeUnit.SECONDS);
+
+            var latestCheckpointOpt =
+                    checkpoints.getHistory().stream()
+                            .filter(
+                                    cp ->
+                                            CheckpointStatsStatus.valueOf(
+                                                            cp.get(
+                                                                            CheckpointStatistics
+                                                                                    .FIELD_NAME_STATUS)
+                                                                    .asText())
+                                                    == CheckpointStatsStatus.COMPLETED)
+                            .filter(
+                                    cp ->
+                                            !cp.get(
+                                                            CheckpointStatistics
+                                                                    .CompletedCheckpointStatistics
+                                                                    .FIELD_NAME_EXTERNAL_PATH)
+                                                    .asText()
+                                                    .equals(
+                                                            NonPersistentMetadataCheckpointStorageLocation
+                                                                    .EXTERNAL_POINTER))
+                            .max(
+                                    Comparator.comparingLong(
+                                            cp ->
+                                                    cp.get(CheckpointStatistics.FIELD_NAME_ID)
+                                                            .asLong()))
+                            .map(
+                                    cp ->
+                                            new Savepoint(
```

Comment:
I am not pretty sure whether we could wrap the checkpoint as the `Savepoint` and store in the `.status.SavepointInfo`. It is a little misleading since savepoint is usually triggered by user manually and is consistent between Flink major versions.

Reply:
I decided to do this because when we record this we are actually in a special scenario. Terminal job, and it's only a checkpoint when the job failed/finished (otherwise it would be a savepoint due to stopwithsavepoint).

I decided to put it in savepointinfo for simplicity from the operator side to avoid introducing new status fields and keeping the logic simple.

The savpoint info in any case is not the real source of truth because anything can happen that prevents us from recording information so I think this is fair. With the savepoint history feature this will be improved further I believe

Reply:
Given that the `.status.jobStatus.savepointInfo` not the real source of truth, it is reasonable to also store the checkpoint into this field.

It will be great if we could also describe this in the documentation :)

Reply:
I suggest we call it lastStateInfo, or add another field for the checkpoint.

Reply:
`lastStateInfo` looks better and then we could have a `snapshotType` field(e.g. checkpoint, savepoint).

Reply:
After an offline discussion with @morhidi we agreed to keep it as is, and work on moving this out from the status to a configmap (after the release) as it's very much internal at this point and not really meant to be for user use.